### ISSUE TITLE
Text/varchar parameters cause "Binder Error: Inconsistent types found for parameter" when used in a combination with bigint column.

### ISSUE THREAD

**Issue by amirrajan** [duckdb/duckdb#4042]
#### What happens?
The following sql, when given the parameters `"value-1"`, and `"value-2"` causes a binding error (reproduction steps goes into details).

##### "The Query"

```sql
-- $1: 'value-1'
-- $2: 'value-2'
SELECT ((SUM(CASE
             WHEN ((stringliterals.a2 = ($1)::text) AND (stringliterals.a3 = ($1)::text))
	     THEN stringliterals.a4
	     ELSE (0)::int
	     END) +
	 SUM(CASE
	     WHEN ((stringliterals.a2 = ($2)::text) AND (stringliterals.a3 = ($2)::text))
	     THEN stringliterals.a4
	     ELSE (0)::int
	     END)) / ((SUM(CASE
	                   WHEN ((stringliterals.a2 = ($1)::text) AND (stringliterals.a3 = ($1)::text))
			   THEN stringliterals.a4
			   ELSE (0)::int
			   END) +
                       SUM(CASE
		           WHEN ((stringliterals.a2 = ($2)::text) AND (stringliterals.a3 = ($2)::text))
			   THEN stringliterals.a4
			   ELSE (0)::int
			   END)))::float8) AS A00000185
FROM stringliterals
```

#### To Reproduce
1. Create `stringliterals.csv` with the following information:
```
a1,a2,a3,a4
1,value-1,value-1,10
```
2. Import the `csv`:
```sql
CREATE TABLE IF NOT EXISTS stringliterals as
SELECT row_number() OVER () AS ID, *
FROM read_csv('./stringliterals.csv', header=1, columns={
  'a1': 'BIGINT',
  'a2': 'VARCHAR',
  'a3': 'VARCHAR',
  'a4': 'BIGINT'
});
``` 
3. Attempt to invoke "The Query" with the parameters `"value-1"`, `"value-2"`:
```ts
type Response = readonly Record<string, unknown>[];
return new Promise<Response>((resolve, reject) => {
  db.all(query, ...params, function (err: Error | null, res: Response) {
    if (err) reject(err);
    else {
      resolve(res);
    }
});
```

The following error occurs: `Binder Error: Inconsistent types found for parameter with index 2`

**Caveat 0**: 
If `"value-1"` and `"value-2"` are inlined (instead of using parameters), then "The Query" **succeeds**.

**Caveat 1**: 
If you replace step 2 with the following (instead of ingesting a csv):
```sql
CREATE TABLE IF NOT EXISTS stringliterals as
select * from (
  values
    (1::int,'value-1'::varchar,'value-1'::varchar,10::int)
) as t(a1,a2,a3,a4);
```
Then "The Query" **succeeds.**

**Caveat 2**:
Here are the `information_schema.columns` for csv import vs inline sql query. 

- Changing the csv import to from `int` instead of `bigint` makes "The Query" **succeed**.
- But, "The Query" **fails** when changing the `ELSE` clause's typecast from `::int` to `::bigint` (while keeping the original csv ingestion unchanged). 
```
┌───────────────┬──────────────┬────────────────┬─────────────┬──────────────────┬────────────────┬─────────────┬───────────┬──────────────────────────┬────────────────────────┬───────────────────┬─────────────────────────┬───────────────┬────────────────────┬───────────────┬────────────────────┬───────────────────────┬──────────────────────┬────────────────────┬───────────────────┬──────────────────┬────────────────┬────────────────┬───────────────┬─────────────┬─────────────┬────────────┬──────────┬───────────────┬──────────────┬────────────┬─────────────────────┬────────────────┬─────────────────────┬─────────────┬─────────────────────┬────────────────┬────────────────────┬──────────────────┬──────────────────┬────────────────┬──────────────┬───────────────────────┬──────────────┐
│ table_catalog │ table_schema │   table_name   │ column_name │ ordinal_position │ column_default │ is_nullable │ data_type │ character_maximum_length │ character_octet_length │ numeric_precision │ numeric_precision_radix │ numeric_scale │ datetime_precision │ interval_type │ interval_precision │ character_set_catalog │ character_set_schema │ character_set_name │ collation_catalog │ collation_schema │ collation_name │ domain_catalog │ domain_schema │ domain_name │ udt_catalog │ udt_schema │ udt_name │ scope_catalog │ scope_schema │ scope_name │ maximum_cardinality │ dtd_identifier │ is_self_referencing │ is_identity │ identity_generation │ identity_start │ identity_increment │ identity_maximum │ identity_minimum │ identity_cycle │ is_generated │ generation_expression │ is_updatable │
├───────────────┼──────────────┼────────────────┼─────────────┼──────────────────┼────────────────┼─────────────┼───────────┼──────────────────────────┼────────────────────────┼───────────────────┼─────────────────────────┼───────────────┼────────────────────┼───────────────┼────────────────────┼───────────────────────┼──────────────────────┼────────────────────┼───────────────────┼──────────────────┼────────────────┼────────────────┼───────────────┼─────────────┼─────────────┼────────────┼──────────┼───────────────┼──────────────┼────────────┼─────────────────────┼────────────────┼─────────────────────┼─────────────┼─────────────────────┼────────────────┼────────────────────┼──────────────────┼──────────────────┼────────────────┼──────────────┼───────────────────────┼──────────────┤
│               │ main         │ stringliterals │ ID          │ 1                │                │ YES         │ BIGINT    │                          │                        │ 64                │ 2                       │ 0             │                    │               │                    │                       │                      │                    │                   │                  │                │                │               │             │             │            │          │               │              │            │                     │                │                     │             │                     │                │                    │                  │                  │                │              │                       │              │
│               │ main         │ stringliterals │ a1          │ 2                │                │ YES         │ BIGINT    │                          │                        │ 64                │ 2                       │ 0             │                    │               │                    │                       │                      │                    │                   │                  │                │                │               │             │             │            │          │               │              │            │                     │                │                     │             │                     │                │                    │                  │                  │                │              │                       │              │
│               │ main         │ stringliterals │ a2          │ 3                │                │ YES         │ VARCHAR   │                          │                        │                   │                         │               │                    │               │                    │                       │                      │                    │                   │                  │                │                │               │             │             │            │          │               │              │            │                     │                │                     │             │                     │                │                    │                  │                  │                │              │                       │              │
│               │ main         │ stringliterals │ a3          │ 4                │                │ YES         │ VARCHAR   │                          │                        │                   │                         │               │                    │               │                    │                       │                      │                    │                   │                  │                │                │               │             │             │            │          │               │              │            │                     │                │                     │             │                     │                │                    │                  │                  │                │              │                       │              │
│               │ main         │ stringliterals │ a4          │ 5                │                │ YES         │ BIGINT    │                          │                        │ 64                │ 2                       │ 0             │                    │               │                    │                       │                      │                    │                   │                  │                │                │               │             │             │            │          │               │              │            │                     │                │                     │             │                     │                │                    │                  │                  │                │              │                       │              │
└───────────────┴──────────────┴────────────────┴─────────────┴──────────────────┴────────────────┴─────────────┴───────────┴──────────────────────────┴────────────────────────┴───────────────────┴─────────────────────────┴───────────────┴────────────────────┴───────────────┴────────────────────┴───────────────────────┴──────────────────────┴────────────────────┴───────────────────┴──────────────────┴────────────────┴────────────────┴───────────────┴─────────────┴─────────────┴────────────┴──────────┴───────────────┴──────────────┴────────────┴─────────────────────┴────────────────┴─────────────────────┴─────────────┴─────────────────────┴────────────────┴────────────────────┴──────────────────┴──────────────────┴────────────────┴──────────────┴───────────────────────┴──────────────┘
┌───────────────┬──────────────┬────────────────┬─────────────┬──────────────────┬────────────────┬─────────────┬───────────┬──────────────────────────┬────────────────────────┬───────────────────┬─────────────────────────┬───────────────┬────────────────────┬───────────────┬────────────────────┬───────────────────────┬──────────────────────┬────────────────────┬───────────────────┬──────────────────┬────────────────┬────────────────┬───────────────┬─────────────┬─────────────┬────────────┬──────────┬───────────────┬──────────────┬────────────┬─────────────────────┬────────────────┬─────────────────────┬─────────────┬─────────────────────┬────────────────┬────────────────────┬──────────────────┬──────────────────┬────────────────┬──────────────┬───────────────────────┬──────────────┐
│ table_catalog │ table_schema │   table_name   │ column_name │ ordinal_position │ column_default │ is_nullable │ data_type │ character_maximum_length │ character_octet_length │ numeric_precision │ numeric_precision_radix │ numeric_scale │ datetime_precision │ interval_type │ interval_precision │ character_set_catalog │ character_set_schema │ character_set_name │ collation_catalog │ collation_schema │ collation_name │ domain_catalog │ domain_schema │ domain_name │ udt_catalog │ udt_schema │ udt_name │ scope_catalog │ scope_schema │ scope_name │ maximum_cardinality │ dtd_identifier │ is_self_referencing │ is_identity │ identity_generation │ identity_start │ identity_increment │ identity_maximum │ identity_minimum │ identity_cycle │ is_generated │ generation_expression │ is_updatable │
├───────────────┼──────────────┼────────────────┼─────────────┼──────────────────┼────────────────┼─────────────┼───────────┼──────────────────────────┼────────────────────────┼───────────────────┼─────────────────────────┼───────────────┼────────────────────┼───────────────┼────────────────────┼───────────────────────┼──────────────────────┼────────────────────┼───────────────────┼──────────────────┼────────────────┼────────────────┼───────────────┼─────────────┼─────────────┼────────────┼──────────┼───────────────┼──────────────┼────────────┼─────────────────────┼────────────────┼─────────────────────┼─────────────┼─────────────────────┼────────────────┼────────────────────┼──────────────────┼──────────────────┼────────────────┼──────────────┼───────────────────────┼──────────────┤
│               │ main         │ stringliterals │ a1          │ 1                │                │ YES         │ INTEGER   │                          │                        │ 32                │ 2                       │ 0             │                    │               │                    │                       │                      │                    │                   │                  │                │                │               │             │             │            │          │               │              │            │                     │                │                     │             │                     │                │                    │                  │                  │                │              │                       │              │
│               │ main         │ stringliterals │ a2          │ 2                │                │ YES         │ VARCHAR   │                          │                        │                   │                         │               │                    │               │                    │                       │                      │                    │                   │                  │                │                │               │             │             │            │          │               │              │            │                     │                │                     │             │                     │                │                    │                  │                  │                │              │                       │              │
│               │ main         │ stringliterals │ a3          │ 3                │                │ YES         │ VARCHAR   │                          │                        │                   │                         │               │                    │               │                    │                       │                      │                    │                   │                  │                │                │               │             │             │            │          │               │              │            │                     │                │                     │             │                     │                │                    │                  │                  │                │              │                       │              │
│               │ main         │ stringliterals │ a4          │ 4                │                │ YES         │ INTEGER   │                          │                        │ 32                │ 2                       │ 0             │                    │               │                    │                       │                      │                    │                   │                  │                │                │               │             │             │            │          │               │              │            │                     │                │                     │             │                     │                │                    │                  │                  │                │              │                       │              │
└───────────────┴──────────────┴────────────────┴─────────────┴──────────────────┴────────────────┴─────────────┴───────────┴──────────────────────────┴────────────────────────┴───────────────────┴─────────────────────────┴───────────────┴────────────────────┴───────────────┴────────────────────┴───────────────────────┴──────────────────────┴────────────────────┴───────────────────┴──────────────────┴────────────────┴────────────────┴───────────────┴─────────────┴─────────────┴────────────┴──────────┴───────────────┴──────────────┴────────────┴─────────────────────┴────────────────┴─────────────────────┴─────────────┴─────────────────────┴────────────────┴────────────────────┴──────────────────┴──────────────────┴────────────────┴──────────────┴───────────────────────┴──────────────┘
```

#### Environment (please complete the following information):
 - OS: MacOS, Linux
 - DuckDB Version/DuckDB Client: DuckDb NodeJS 0.4.0

#### Identity Disclosure:
 - Full Name: Amir Rajan
 - Affiliation: Mode Analytics

If the above is not given and is not obvious from your GitHub profile page, we might close your issue without further review. Please refer to the [reasoning behind this rule](https://berthub.eu/articles/posts/anonymous-help/) if you have questions.

#### Before Submitting
- I'll see if I can get you a runnable program that reproduces the issue. (verified against `master/453d09706dad8ccafff849891c568597f0fb073d`)

**Comment by Mytherin** [comment]
This should be resolved by #4059 and seems to work for me in the master

**Comment by amirrajan** [comment]
Thanks! I'll give it a shot locally
